### PR TITLE
Add command /viewbanlist

### PR DESCRIPTION
**Pull request by MacChaeger**
This command is intended to be a quick-access option for checking banlists and rulesets for formats as well as descriptions of the rules that would get listed in case the user has questions.

### CODE REVIEW

**Comment by AustinXII** [comment]
Ah I see what you meant, this is good :+1

**Comment by Zarel** [comment]
Isn't there already an "about format" command? This should just go into that.

**Comment by MacChaeger** [comment]
Well, I was going to do that, but I realized I would need to look up how the difference between `/data` and `/details` works, and then I forgot to actually do that.

File: data/rulesets.js [hunk 1]
```diff
@@ -11,24 +11,28 @@ exports.BattleFormats = {
...
 		effectType: 'ValidatorRule',
 		name: 'Standard NEXT',
+		desc: ["The standard ruleset for the NEXT mod"],
 		ruleset: ['Sleep Clause Mod', 'Species Clause', 'Nickname Clause', 'OHKO Clause', 'HP Percentage Mod', 'Cancel Mod'],
 		banlist: ['Illegal', 'Soul Dew'],
 	},
 	standardubers: {
 		effectType: 'ValidatorRule',
 		name: 'Standard Ubers',
+		desc: ["The standard ruleset for [Gen 5] Ubers"],
```
Comment (Zarel) [inline data/rulesets.js]: `[Gen 5]`???

Comment (MacChaeger) [reply data/rulesets.js]: Yeah, the only format that uses this list is `[Gen 5] Ubers`

Comment (Zarel) [reply data/rulesets.js]: That makes sense... should probably get rid of it, then?

File: data/rulesets.js [hunk 2]
```diff
@@ -374,6 +387,7 @@ exports.BattleFormats = {
 	abilityclause: {
```
Comment (Zarel) [inline data/rulesets.js]: Should possibly be named Hackmons Ability Clause.

Comment (MacChaeger) [reply data/rulesets.js]: AAA uses it too

Comment (Zarel) [reply data/rulesets.js]: Two Ability Clause?

Comment (MacChaeger) [reply data/rulesets.js]: Yeah, AAA and BH and both use the same ability clause.

File: data/rulesets.js (hunk 1 above)
Comment (MacChaeger) [reply data/rulesets.js]: Should that format just use `'Standard'` with `'Evasion Moves Clause'` as a removed rule, then, or did you have something else in mind?

File: chat-plugins/info.js [hunk 3]
```diff
@@ -1407,6 +1429,18 @@ exports.commands = {
 		return this.sendReply("|raw|" + buf.join("") + "");
 	},
 
+	'!viewbanlist': true,
+	viewruleset: 'viewbanlist',
+	ruleset: 'viewbanlist',
+	banlist: 'viewbanlist',
+	viewbanlist: function (target, room, user, connection, cmd) {
+		if (!target) {
+			return this.sendReply("Usage: " + cmd + " <format>");
```
Comment (panpawn) [inline chat-plugins/info.js]: This will say:
``Usage: viewbanlist <format>``

I think you forgot the ``/`` in the UI. Also, new code should probably be using template strings.

File: chat-plugins/info.js [hunk 4]
```diff
@@ -1378,11 +1379,32 @@ exports.commands = {
 
 		if (!totalMatches) return this.errorReply("No " + (target ? "matched " : "") + "formats found.");
 		if (totalMatches === 1) {
+			let html = [];
+			if (isBanlistSearch) {
+				let format = Dex.getFormat(targetId);
+				if (format.effectType === 'ValidatorRule' || format.effectType === 'Rule' || format.effectType === 'Format') {
+					if (format.ruleset && format.ruleset.length) html.push("<b>Ruleset</b> - " + Chat.escapeHTML(format.ruleset.join(", ")));
+					//if (format.removedRules && format.removedRules.length) html.push("<b>Removed rules</b> - " + Chat.escapeHTML(format.removedRules.join(", "))); // No current formats use this, but just in case...
```
Comment (panpawn) [inline chat-plugins/info.js]: Use a single space after ``//`` comments

File: chat-plugins/info.js [hunk 5]
```diff
@@ -1344,6 +1344,7 @@ exports.commands = {
 		}
 
 		const isOMSearch = (cmd === 'om' || cmd === 'othermetas');
+		const isBanlistSearch = (cmd in ['viewruleset', 'viewbanlist', 'ruleset', 'banlist']);
```
Comment (panpawn) [inline chat-plugins/info.js]: ... I'm pretty sure this isn't a thing? ?_?

```Javascript
const banlistSearches = ['viewruleset', 'viewbanlist', 'ruleset', 'banlist'];
const isBanlistSearch = banlistSearches.includes(cmd);
```

File: data/rulesets.js (hunk 2 above)
Comment (Zarel) [reply data/rulesets.js]: No, I mean, how about naming it "Two Ability Clause"?

File: chat-plugins/info.js (hunk 5 above)
Comment (MacChaeger) [reply chat-plugins/info.js]: Oh yeah

File: data/rulesets.js (hunk 2 above)
Comment (MacChaeger) [reply data/rulesets.js]: @TheImmortal I'd say this is up to you

**Comment by Zarel** [comment]
I still think this should be inside `/formathelp`

**Comment by MacChaeger** [comment]
I... thought this was what you meant.

File: data/rulesets.js (hunk 2 above)
Comment (TheImmortal) [reply data/rulesets.js]: Ability Clause is fine imo.

**Comment by Zarel** [comment]
I mean just show it, in a collapsible box, normally?

**Comment by MacChaeger** [comment]
Okay, well, I'd have to figure out how to do that since I don't think there are any existing commands that have a collapsible box which I could copy the code for. With that said, what command should be used if someone wants data on a ruleset? `/formathelp` used to only look through the official tiers unless it was called with `/om` or `/othermetas`. I'm assuming you'd want something like that for searching through rulesets, but in that case I'm not totally sure what the command should be named.

**Comment by Zarel** [comment]
Something like just `/format` should be an alias for `/formathelp` if it isn't already...

**Comment by MacChaeger** [comment]
Have I mentioned I don't know html?

**Comment by Zarel** [comment]
Cool, it's rare for a person to know a programming language but not HTML.

HTML is a markup language, mainly what it does is let you put stuff in tags, like:

```html
<p>This is a <strong>paragraph</strong>.</p>
<p>This is <em>another</em> paragraph.</p>
```

Should be mostly comprehensible from just that, honestly. Expand/collapse sections are used for other parts of the code so in theory you can just copy those.

File: data/rulesets.js (hunk 1 above)
Comment (Zarel) [reply data/rulesets.js]: Yes, do that.

**Comment by MacChaeger** [comment]
Oh, that was a disclaimer for my current attempt at that, not an excuse for not doing it. I just cobbled together some code from multiple sources and hoped the syntax worked, but I didn't know there was any collapsible text anywhere else on PS.

**Comment by Zarel** [comment]
@panpawn you've done stuff with it, right? can you help?

**Comment by MacChaeger** [comment]
Where is that collapsible text, exactly?

**Comment by Zarel** [comment]
I don't actually remember; I think it's in one of panpawn's pullreqs though so I asked him

**Comment by panpawn** [comment]
For a simple example, it's used in the ``!code`` command: https://github.com/Zarel/Pokemon-Showdown/blob/master/chat-commands.js#L142

File: config/formats.js [hunk 6]
```diff
@@ -960,7 +960,8 @@ exports.Formats = [
 		],
 
 		mod: 'gen5',
-		ruleset: ['Pokemon', 'Team Preview', 'Standard Ubers'],
+		ruleset: ['Pokemon', 'Team Preview', 'Standard'],
+		removedRules: ['Evasion Moves Clause'],
```
Comment (sirDonovan) [inline config/formats.js]: <code>removedRules</code> is not a valid format property. It's only used in the tournaments code.

Comment (MacChaeger) [reply config/formats.js]: Yeah I was gonna look into that but it must have slipped my mind. Direct me to the right file?

Comment (sirDonovan) [reply config/formats.js]: There is no file to direct you to in this case. It's only used as a local variable in displaying custom rules for tournaments.

Comment (MacChaeger) [reply config/formats.js]: I meant to the file where I can add in that functionality.

File: chat-plugins/info.js [hunk 7]
```diff
@@ -1383,16 +1383,11 @@ exports.commands = {
...
 				if (format.banlist && format.banlist.length) html.push("<b>Bans</b> - " + Chat.escapeHTML(format.banlist.join(", ")));
 				if (format.unbanlist && format.unbanlist.length) html.push("<b>Unbans</b> - " + Chat.escapeHTML(format.unbanlist.join(", ")));
 				if (html.length > 0) {
-					html.unshift(
-						"<b>Ruleset for " + format.name + ":</b><br />" +
-						"<a href=\"#rules\" class=\"fa fa-list-ul\" data-toggle=\"collapse\">Ruleset/Banilst</a><br />" +
-						"<div id=\"rules\" class=\"collapse\">"
-					);
-					html.push("</div>");
+					html = `<details><summary>Banilist/Ruleset</summary>${html.join("<br />")}</details>`;
```
Comment (panpawn) [inline chat-plugins/info.js]: "Banilist"?

File: config/formats.js (hunk 6 above)
Comment (sirDonovan) [reply config/formats.js]: <code>sim/dex.js</code> and <code>sim/dex-data.js</code>, but you should probably do that in a separate PR.

**Comment by MacChaeger** [comment]
Ok i'm gonna get rid of `'Standard Ubers'` in another PR.

File: config/formats.js (hunk 6 above)
Comment (Zarel) [reply config/formats.js]: I think this should be best done with:

`ruleset: ['Pokemon', 'Team Preview', 'Standard', '!Evasion Moves Clause'],`

**Comment by MacChaeger** [comment]
Other than the whole standard ubers thing, is there anything missing here?

File: data/rulesets.js (hunk 2 above)
Comment (Zarel) [reply data/rulesets.js]: I soft-veto "Ability Clause". Pick a better name.

File: chat-plugins/info.js [hunk 8]
```diff
@@ -1378,11 +1379,30 @@ exports.commands = {
 
 		if (!totalMatches) return this.errorReply("No " + (target ? "matched " : "") + "formats found.");
 		if (totalMatches === 1) {
-			let format = Dex.getFormat(Object.values(sections)[0].formats[0]);
+			let html = [];
+			let format = Dex.getFormat(targetId);
```
Comment (urkerab) [inline chat-plugins/info.js]: Why did you change the value of `format` here? If you get a single match but it's not an exact match then you get a message about there being no banlist/ruleset.

Comment (MacChaeger) [reply chat-plugins/info.js]: I was copy/pasting code from when it was a separate command. Should I just change it back?

Comment (urkerab) [reply chat-plugins/info.js]: I think it would make sense to change it back, yes.
